soc/arm/atmel_sam0/common/atmel_sam0_dt.h

```diff
@@ -47,4 +48,55 @@
 #define ATMEL_SAM0_DT_CPU_CLK_FREQ_HZ \
 	DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency)
 
+/* Devicetree related macros to construct pin mux config data */
+
+/* Get a node id from a pinctrl-0 prop at index 'i' */
+#define NODE_ID_FROM_PINCTRL_0(inst, i) \
+	DT_INST_PHANDLE_BY_IDX(inst, pinctrl_0, i)
```

Comment:
On Atmel SAM family the macros were recently converted to use `node_id` and not `inst` as a parameter #36213. This allows for the macros to be used outside the context of a driver. Will be used by the future pinctrl driver. We could do the same here.